crates/sui-framework/packages/dwallet-system/sources/ethereum_state.move

```diff
@@ -45,6 +48,7 @@ module dwallet_system::ethereum_state {
             id: object::new(ctx),
             data: state_data,
             time_slot: 0u64,
+            state_root: vector::empty(),
```

Comment:
No no, the purpose of the `init` function is to initialize to a trusted starting point from which verification should occur. This means that there should be a value here for the starting point `state_root` which points to some block in Ethereum that we have as a sort of "later genesis" we trust. Same on `time_slot`.

Reply:
There is absolutely no use for the `state_root` in the state verification process.
The only purpose of this field is to have it later when we verify a message.

Let's say that this is the first time we use the initialized state object.
The first thing that would be done is bootstrapping the state using the `checkpoint` only (later we can add sync committee and some more data to initializer function, then we would not need to bootstrap).
Later, after verifying the state, we keep the updated `state_root` and `time_slot` in these fields, and use them later for the message verification.

Reply:
In this case, please in this function also call `verify_new_state`. I don't want to have unused, uninitialized values.
An alternative is to say we will remove this function and create these objects in Genesis, in which case we should add a `TODO` here and document this explicit in the issue. (Link issue in `TODO`)

Reply:
this is not the case anymore, i've changed the function to also return the state_root and set it to the correct one.